external-builds/pytorch/setup_venv.sh

```diff
@@ -10,11 +10,24 @@ VENV_DIR="${1:-$(realpath "$(dirname "${BASH_SOURCE[0]}")/../..")/venv}"
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 REQUIREMENTS_FILE="${SCRIPT_DIR}/requirements-test.txt"
 
-# Create virtual environment
+# Ensure INDEX_URL is set
+if [[ -z "${INDEX_URL:-}" ]]; then
+  echo "ERROR: INDEX_URL environment variable must be set"
+  exit 1
+fi
+
+# Ensure TORCH_VERSION is set
+if [[ -z "${TORCH_VERION:-}" ]]; then
+  echo "ERROR: TORCH_VERSION environment variable must be set"
+  exit 1
+fi
```

Comment:
I don't think these checks are needed, given that the script uses `set -u`: https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#set--u

You can also expand the variable with `?` to check at the time of expansion: https://stackoverflow.com/questions/8889302/what-is-the-meaning-of-a-question-mark-in-bash-variable-parameter-expansion-as-i

The error messages are nice... but we should be replacing this script with a python script anyways.

Reply:
(needing to understand all these bash gotchas is a big reason to move to Python)